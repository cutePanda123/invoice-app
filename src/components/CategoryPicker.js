import React from 'react';
import PropTypes from 'prop-types';
import Ionicon from 'react-ionicons';

class CategoryPicker extends React.Component {
    constructor(props) {
        super(props);
    }

    selectCategory = (event, category) => {
        this.props.onSelectCategory(category);
        event.preventDefault();
    }

    render() {
        const { categories, selectedCategory } = this.props;
        const selectedCategoryId = selectedCategory && selectedCategory.id;
        return (
            <div className="category-picker-component">
                <div className="row">
                    {
                        categories.map((category, index) => {
                            const iconColor = (category.id === selectedCategoryId) ? '#fff' : '#555';
                            const backColor = (category.id === selectedCategoryId) ? '#347eff' : '#efefef';
                            const activeClassName = selectedCategoryId == category.id ? 
                                "category-item col-3 active" : "category-item col-3";
                            return (
                                <div
                                    className={activeClassName}
                                    key={index}
                                    onClick={(event) => {
                                        this.selectCategory(event, category);
                                    }}
                                    role='button'
                                    style={{textAlign: 'center'}}
                                >
                                    <Ionicon 
                                        className="rounded-circle"
                                        style={
                                            {
                                                backgroundColor: backColor,
                                                padding: '5px'
                                            }
                                        }
                                        fontSize="50px"
                                        color={iconColor}
                                        icon={category.iconName}
                                    />
                                    <p>{category.name}</p>
                                </div>
                            );
                        })
                    }
                </div>
            </div>
        );
    }
}

CategoryPicker.propTypes = {
    categories: PropTypes.array.isRequired,
    onSelectCategory: PropTypes.func.isRequired,
    selectedCategory: PropTypes.object
};

export default CategoryPicker;

