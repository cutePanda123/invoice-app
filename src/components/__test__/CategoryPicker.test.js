import React from 'react';
import { mount } from 'enzyme';
import CategoryPicker from '../CategoryPicker';
import Ionicon from 'react-ionicons';

export const categories = [
    {
      "id": 1,
      "name": "travel",
      "type": "outcome",
      "iconName": "ios-plane"
    },
    {
      "id": 2,
      "name": "investement",
      "type": "income",
      "iconName": "logo-yen"
    }
];

let props = {
    categories,
    onSelectCategory: jest.fn()
};

let propsWithSelectedCategory = {
    categories,
    onSelectCategory: jest.fn(),
    selectedCategory: categories[0]
};
let wrapper = null;

describe('CategoryPicker test', () => {
    it('snapshot test', () => {
        wrapper = mount(<CategoryPicker {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render should have correct number of items', () => {
        wrapper = mount(<CategoryPicker {...props} />);
        expect(wrapper.find('.category-item').length).toEqual(categories.length);
        expect(wrapper.find('.category-item.active').length).toEqual(0);
        const firstIcon = wrapper.find('.category-item').first().find(Ionicon);
        expect(firstIcon.length).toEqual(1);
        expect(firstIcon.props().icon).toEqual(categories[0].iconName);
    });

    it('render with selected category should have an active item', () => {
        wrapper = mount(<CategoryPicker {...propsWithSelectedCategory} />);
        expect(wrapper.find('.category-item.active').length).toEqual(1);
    });

    it('select a category should trigger callback and change the active item', () => {
        wrapper = mount(<CategoryPicker {...propsWithSelectedCategory} />);
        wrapper.find('.category-item').at(1).simulate('click');
        expect(propsWithSelectedCategory.onSelectCategory).toHaveBeenCalledWith(categories[1]);
    });
});