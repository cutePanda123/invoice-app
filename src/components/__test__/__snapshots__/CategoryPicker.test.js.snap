// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPicker test snapshot test 1`] = `
<CategoryPicker
  categories={
    Array [
      Object {
        "iconName": "ios-plane",
        "id": 1,
        "name": "travel",
        "type": "outcome",
      },
      Object {
        "iconName": "logo-yen",
        "id": 2,
        "name": "investement",
        "type": "income",
      },
    ]
  }
  onSelectCategory={[MockFunction]}
>
  <div
    className="category-picker-component"
  >
    <div
      className="row"
    >
      <div
        className="category-item col-3"
        key="0"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <Ionicon
          beat={false}
          className="rounded-circle"
          color="#555"
          fontSize="50px"
          icon="ios-plane"
          rotate={false}
          shake={false}
          style={
            Object {
              "backgroundColor": "#efefef",
              "padding": "5px",
            }
          }
        >
          <styled.svg
            beat={0}
            className="rounded-circle"
            fill="#555"
            height="50px"
            rotate={0}
            shake={0}
            style={
              Object {
                "backgroundColor": "#efefef",
                "padding": "5px",
              }
            }
            viewBox="0 0 1024 1024"
            width="50px"
          >
            <svg
              className="rounded-circle sc-bdVaJa fUuvxv"
              fill="#555"
              height="50px"
              rotate={0}
              style={
                Object {
                  "backgroundColor": "#efefef",
                  "padding": "5px",
                }
              }
              viewBox="0 0 1024 1024"
              width="50px"
            >
              <path
                d="M448 144h-96l118 311.4-253.6 8.6-88.4-112h-64l64 160-64 160h64l89.8-115.4 252.2 10.4-118 313h96l196.4-309 135.6 3c6.6 0.4 13.2 0.2 20 0.2 88.4 0 160-28.8 160-64.2s-71.6-64-160-64c-6.8 0-13.4 0.2-20 0.4l-135.6 4.4-196.4-306.8z"
              />
            </svg>
          </styled.svg>
        </Ionicon>
        <p>
          travel
        </p>
      </div>
      <div
        className="category-item col-3"
        key="1"
        onClick={[Function]}
        role="button"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <Ionicon
          beat={false}
          className="rounded-circle"
          color="#555"
          fontSize="50px"
          icon="logo-yen"
          rotate={false}
          shake={false}
          style={
            Object {
              "backgroundColor": "#efefef",
              "padding": "5px",
            }
          }
        >
          <styled.svg
            beat={0}
            className="rounded-circle"
            fill="#555"
            height="50px"
            rotate={0}
            shake={0}
            style={
              Object {
                "backgroundColor": "#efefef",
                "padding": "5px",
              }
            }
            viewBox="0 0 1024 1024"
            width="50px"
          >
            <svg
              className="rounded-circle sc-bdVaJa fUuvxv"
              fill="#555"
              height="50px"
              rotate={0}
              style={
                Object {
                  "backgroundColor": "#efefef",
                  "padding": "5px",
                }
              }
              viewBox="0 0 1024 1024"
              width="50px"
            >
              <path
                d="M896 64h-160l-224 442.256-224-442.256h-160l224.736 416h-96.736v96h147.128l28.872 62v34h-176v96h176v192h160v-192h176v-96h-176v-34l29.782-62h146.218v-96h-96.578l224.578-416z"
              />
            </svg>
          </styled.svg>
        </Ionicon>
        <p>
          investement
        </p>
      </div>
    </div>
  </div>
</CategoryPicker>
`;
